src/middleware/jiraSymmetricJwtMiddleware.ts

```diff
@@ -0,0 +1,140 @@
+import { decodeSymmetric, getAlgorithm } from "atlassian-jwt";
+import { NextFunction, Request, Response } from "express";
+import { booleanFlag, BooleanFlags } from "~/src/config/feature-flags";
+import { TokenType, verifyQsh } from "~/src/jira/util/jwt";
+import { Installation } from "~/src/models/installation";
+import { moduleUrls } from "~/src/routes/jira/atlassian-connect/jira-atlassian-connect-get";
+import { matchRouteWithPattern } from "~/src/util/match-route-with-pattern";
+
+export const jiraSymmetricJwtMiddleware = async (req: Request, res: Response, next: NextFunction) => {
+
+	if (!await booleanFlag(BooleanFlags.NEW_JWT_VALIDATION, false)) {
+		return next();
+	}
+
+	const token = req.query?.["jwt"] || req.cookies?.["jwt"];
+
+	if (token) {
+		const issuer = getIssuer(token);
+		if (!issuer) {
+			req.log.info("JWT claim did not contain the issuer (iss) claim");
+			return res.status(401).send("Unauthorised");
+		}
+
+		const installation = await Installation.getForClientKey(issuer);
+		if (!installation) {
+			req.log.info("No Installation found");
+			return res.status(401).send("Unauthorised");
+		}
+
+		const secret = await installation.decrypt("encryptedSharedSecret");
+
+		const path = req.baseUrl == "" ? req.path : req.baseUrl;
+		const tokenType = checkPathValidity(path) ? TokenType.normal : TokenType.context;
+		try {
+			verifySymmetricJwt(token, secret, req, tokenType);
+		} catch (err) {
+			req.log.info(err.message);
```

Comment:
Same: I feel like this should be a warn. Also, would it make sense to add error as the first arg?
```
catch (err) {
req.log.warn({ err }, "Could not verify symmetric JWT");
}
```